include/fmt/core.h

```diff
@@ -902,20 +909,37 @@ class basic_parse_context : private ErrorHandler {
   FMT_CONSTEXPR unsigned next_arg_id();
 
   FMT_CONSTEXPR bool check_arg_id(unsigned) {
-    if (next_arg_id_ > 0) {
+    if (is_indexing_arguments_automatically()) {
       on_error("cannot switch from automatic to manual argument indexing");
       return false;
     }
     next_arg_id_ = -1;
     return true;
   }
-  void check_arg_id(basic_string_view<Char>) {}
+
+  FMT_CONSTEXPR bool check_arg_id(internal::auto_id) {
```

Comment:
I don't think it is a good idea to add this implementation detail to the public API. `next_arg_id` is there for automatic indexing.

Reply:
The `dynamic_specs_handler` has two template methods to check arguments. I wanted to avoid adding an explicit `on_dynamic_width(auto_id)` and `on_dynamic_precision(auto_id)` overloads. But yes, you're right. Keeping the public API clean is more important (in fact, I didn't realize that it'll be exposed to the public API). I'll change it.